### PR TITLE
Extend high merchandising report

### PR DESCRIPTION
**Pull request by Calum-Campbell**
## What does this change?

Extends the high merchandising report so that adUnits and the edition targeted(if set in DFP) are reported.
This will be used in extending the server side check to decide wether to render the commercial-high slot, which will be done in a separate PR.
## What is the value of this and can you measure success?

Will eventually aid in reducing the number of empty ad units served.
## Does this affect other platforms - Amp, Apps, etc?

No
## Request for comment

@kelvin-chappell @JonNorman @rich-nguyen

### CODE REVIEW

File: common/app/common/dfp/TagSponsorship.scala [hunk 1]
```diff
@@ -91,7 +91,21 @@ object HighMerchandisingLineItems {
...
+case class HighMerchandisingLineItem(
+  name: String,
+  id: Long,
+  tags: Seq[String],
+  adUnits: Seq[GuAdUnit],
+  customTargetSet: Seq[CustomTargetSet]
+  ) {
+  val adUnitIds = adUnits.map(adUnit => adUnit.id)
+  val adUnitString = adUnits.map(adUnit =>
+    adUnit.path.foldLeft(""){(theGuardian, tail) => theGuardian.drop(1) +"/" + tail})
```
Comment (JonNorman) [inline common/app/common/dfp/TagSponsorship.scala]: Could this be simplified to `adUnit.path.tail.mkString("/")`?

File: common/app/common/dfp/TagSponsorship.scala [hunk 2]
```diff
@@ -91,7 +91,21 @@ object HighMerchandisingLineItems {
...
 
-case class HighMerchandisingLineItem(name: String, id: Long, tags: Seq[String])
+case class HighMerchandisingLineItem(
+  name: String,
+  id: Long,
+  tags: Seq[String],
+  adUnits: Seq[GuAdUnit],
+  customTargetSet: Seq[CustomTargetSet]
+  ) {
+  val adUnitIds = adUnits.map(adUnit => adUnit.id)
```
Comment (JonNorman) [inline common/app/common/dfp/TagSponsorship.scala]: Can use the syntactic sugar of `adUnits map (_.id)`

File: common/app/common/dfp/TagSponsorship.scala [hunk 3]
```diff
@@ -91,7 +91,21 @@ object HighMerchandisingLineItems {
...
+  val adUnitString = adUnits.map(adUnit =>
+    adUnit.path.foldLeft(""){(theGuardian, tail) => theGuardian.drop(1) +"/" + tail})
+  val customTargets = customTargetSet.map(set =>
+    set.targets
```
Comment (JonNorman) [inline common/app/common/dfp/TagSponsorship.scala]: As with the `adUnit.id` above

**Comment by JonNorman** [comment]
Apart from minor comments, looks good! :+1:

File: admin/app/views/commercial/highMerchandisingTargetedTags.scala.html [hunk 4]
```diff
@@ -31,7 +31,9 @@
                         <td>@for(adUnitId <- lineItem.adUnitIds) {
                             <a href="@DfpLink.adUnit(adUnitId)">@{adUnitId}</a>
                         }</td>
-                        <td>@{lineItem.editionIfSet.distinct}</td>
+                        @if(lineItem.editionIfSet.isEmpty){
```
Comment (JonNorman) [inline admin/app/views/commercial/highMerchandisingTargetedTags.scala.html]: If `editionIfSet` is just a list of editions against the line item then you can probably just rename it to `editions`; if it is `Nil` then it is fairly clear that there is no editional targeting on the line item. Likewise it is probably clear enough to leave a blank cell there when there is no targeting. Up to you!

**Comment by Calum-Campbell** [comment]
@JonNorman cheers, refactored now to make more concise.

**Pull request merged**